test/case/convention/openshift/golang-osd-operator/07-csv-generate-common

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash -x
+
+set_saas_git_config() {
+  # This is to avoid git failure when running within the docker container
+  # Container doesn't have global config so complaining for missing user/email
+  pushd saas-file-generate-bundle
+  git config user.name "Test Boilerplate" 
+  git config user.email "test@example.com" 
+  popd
+   
+}
+
+# Scenario : 
+# 1 - Create a new CSV and push it to Prod and Staging
+# 2 - Ensure proper history in case of unpushed changed
+# 3 - Validate behaviour when some versions are not pushed to production (history cleanup)
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+
+source $REPO_ROOT/test/lib.sh
+
+repo=$(empty_repo)
+saas_repo=$(empty_repo)
+
+add_cleanup $repo
+add_cleanup $saas_repo
+
+test_project="file-generate"
+
+bootstrap_project $repo ${test_project} openshift/golang-osd-operator
+sed -i '1s,^,include project.mk\n,' $repo/Makefile
+
+# Init the saas git project
+saas_root_repo=$(mktemp -d -t boilerplate-saas-XXXXXXXX)
+cp -R $REPO_ROOT/test/projects/saas-file-generate-bundle-new $saas_root_repo/saas-file-generate-bundle
+saas_repo=${saas_root_repo}/saas-file-generate-bundle
+pushd $saas_repo 
+git init 
+git config user.name "Test Boilerplate" 
+git config user.email "test@example.com" 
+git add .
+git commit -m "Initial saas-file-generate-bundle"
+git remote add origin git@example.com:example-org/test-repo.git
+
+git branch staging master
+git branch production master
+
+export GIT_PATH=`pwd`
+popd
+
+# export DEPLOYED_HASH="0.1.0-441cd20"
+
+cd $repo
+
+make boilerplate-update
+
+colorprint $ORANGE "First build and compare on Staging"
+make staging-common-csv-build
+
+
+if [ ! -f saas-file-generate-bundle/file-generate/1.0.1-*/file-generate.v1.0.1-*  ] ; then 
+    exit 1
+fi
+
+grep "replaces:" saas-file-generate-bundle/file-generate/1.0.1-*/file-generate.v1.0.1-* 
+
+if [ $? -ne 1 ] ; then 
+    exit 1
+fi
+
+previous_version=`ls saas-file-generate-bundle/file-generate | grep "^1.0.0"`
+
+colorprint $ORANGE "First push on Staging"
+set_saas_git_config
+make staging-saas-bundle-push
+
+# ToDo : Check there is a new commit on staging branch and it contains the previous_version CSV
+
+colorprint $ORANGE "First build and compare on Production"
+make production-common-csv-build
+
+colorprint $ORANGE "First push on Production"
+set_saas_git_config
+make production-saas-bundle-push
```

Comment:
(throughout) Every time I see one of these `*-push` targets, my brain interprets the `push` as meaning that we're actually pushing to quay -- which of course we don't want to do in a test. I had to go read code to figure out that you meant "push to git origin", and you're using "publish" to mean "push to quay". This is further confused by the fact that `catalog-publish.sh` uses `push_catalog` as the variable name for the "publish" action.

What would make it less confusing IMO is:
- Instead of having to add an option to make the script push, instead use an option (via an env var) to _skip_ the push. That way lines like this could say e.g. `make DRY_RUN=true production-saas-bundle-push`. 
- Always use "push" to refer to the quay side, leaving `git push` as implicit, since you're dry-running that part internally by overriding the `origin` remote.